src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -7134,14 +7136,179 @@ std::pair<MergeTreeData::MutableDataPartPtr, scope_guard> MergeTreeData::cloneAn
     if (params.copy_instead_of_hardlink)
         with_copy = " (copying data)";
 
-    auto dst_part_storage = src_part_storage->freeze(
-        relative_data_path,
-        tmp_dst_part_name,
-        read_settings,
-        write_settings,
-        /* save_metadata_callback= */ {},
-        params);
+    std::shared_ptr<IDataPartStorage> dst_part_storage{};
+    try
+    {
+        dst_part_storage = src_part_storage->freeze(
+            relative_data_path,
+            tmp_dst_part_name,
+            read_settings,
+            write_settings,
+            /* save_metadata_callback= */ {},
+            params);
+    }
+    catch (...)
```

Comment:
I'd prefer if we check for particular exception codes.

Reply:
There may be some filesystem exceptions such as std::filesystem::filesystem_error.  However, I think we don't have to due with these exceptions because if freeze fails, we will start the copy branch. If you have any other suggestions please let me know and thank you for your time.